Documentation/Components.md

```diff
@@ -0,0 +1,246 @@
+# Babylon Native Components
+
+This document gives a high-level overview of the components provided by 
+default in the Babylon Native repository. For information about the 
+definition of _component_ and the different categories of components, 
+please read the page on the 
+[Babylon Native build system](BuildSystem.md). For information about how to 
+create new components and add or reference them in Babylon Native, please
+read the page on 
+[extending Babylon Native](Extending.md).
+
+## Dependencies
```

Comment:
This seems like it should be merged with https://github.com/BabylonJS/BabylonNative#development-notes?

Reply:
We might could do so to some degree, but in my mind those two pieces of documentation serve meaningfully different purposes. The section in the README is, from what I recall, something along the lines of an "external credits" list intended to call out 3rd party dependencies we felt should be given particular notice. (It may be woefully out-of-date for this purpose.) `Components.md`, on the other hand, I specifically wrote to be a touchpoint for new Babylon Native developers who just built their first BabylonNative.sln and want some high-level information information about what all these components they're seeing are, regardless of the origin and nature of those components. Consequently, things like base-n show up in the README but not here since it doesn't show up as a component (it's a header-only implementation detail of UrlLib, I think) while things like UrlLib show up here but nowhere else because it's not external. `Components.md` also goes out of its way to clarify several cases where things are and aren't part of externally-exposed contracts. For these reasons, I'm inclined to keep these lists distinct so that they can serve separate purposes. I don't think it's hugely important, though, so I'm open to merging them if you feel strongly that that's a better thing to do.

Reply:
> "external credits" list intended to call out 3rd party dependencies we felt should be given particular notice

I didn't write it to be that. I wrote it as a call-out to certain things that were happening in the code. I think if you move some of the info into your section, the README section would be unnecessary.

> base-n

This is used by the Window polyfill for `atob`. I don't think we need to call this out at all.